includes/UserMailer.php

```diff
@@ -434,114 +412,189 @@ protected static function quotedPrintableCallback( $matches ) {
-			/* RT#55604: Add a timeout to the watchlist email block */
-			global $wgEnableWatchlistNotificationTimeout, $wgWatchlistNotificationTimeout;
-			if ( !empty( $otherParam['notisnull'] ) ) {
-				$notificationTimeoutSql = "1";
-			} elseif ( !empty($wgEnableWatchlistNotificationTimeout) && isset($wgWatchlistNotificationTimeout) ) {
-				$blockTimeout = wfTimestamp( TS_MW, wfTimestamp( TS_UNIX, $timestamp ) - intval( $wgWatchlistNotificationTimeout ) );
-				$notificationTimeoutSql = "wl_notificationtimestamp IS NULL OR wl_notificationtimestamp < '$blockTimeout'";
-			} else {
-				$notificationTimeoutSql = 'wl_notificationtimestamp IS NULL';
-			}
-			/* Wikia change end */
-
-			$dbw = wfGetDB( DB_MASTER );
-			$res = $dbw->select( array( 'watchlist' ),
-				array( 'wl_user' ),
-				array(
-					'wl_title' => $title->getDBkey(),
-					'wl_namespace' => $title->getNamespace(),
-					'wl_user != ' . intval( $editor->getID() ),
-					$notificationTimeoutSql,
-				), __METHOD__
-			);
-			foreach ( $res as $row ) {
-				$watchers[] = intval( $row->wl_user );
-			}
+			$notificationTimeoutSql = $this->getTimeOutSql();
+			$watchers = $this->getWatchersToNotify( $notificationTimeoutSql );
 			if ( $watchers ) {
-				// Update wl_notificationtimestamp for all watching users except
-				// the editor
-				$wl = WatchedItem::fromUserTitle( $editor, $title );
-				$wl->updateWatch( $watchers, $timestamp );
+				$this->updateWatchedItem( $watchers );
 			}
+			wfRunHooks( 'NotifyOnSubPageChange', [ $watchers, $this->title, $this->editor, $notificationTimeoutSql ] );
+		}
+		if ( $this->shouldSendEmail( $watchers ) ) {
+			$this->actuallyNotifyOnPageChange( $watchers );
+		}
+	}
 
-			/* Wikia change begin - @author: Jakub Kurcek */
-			wfRunHooks( 'NotifyOnSubPageChange', array ( $watchers, $title, $editor, $notificationTimeoutSql ) );
-			/* Wikia change end */
-		}
-
-		$sendEmail = true;
-		// If nobody is watching the page, and there are no users notified on all changes
-		// don't bother creating a job/trying to send emails
-		// $watchers deals with $wgEnotifWatchlist
-		if ( !count( $watchers ) && !count( $wgUsersNotifiedOnAllChanges ) ) {
-			$sendEmail = false;
-			// Only send notification for non minor edits, unless $wgEnotifMinorEdits
-			if ( !$minorEdit || ( $wgEnotifMinorEdits && !$editor->isAllowed( 'nominornewtalk' ) ) ) {
-				$isUserTalkPage = ( $title->getNamespace() == NS_USER_TALK );
-				if ( $wgEnotifUserTalk && $isUserTalkPage && $this->canSendUserTalkEmail( $editor, $title, $minorEdit ) ) {
-					$sendEmail = true;
-				}
-			}
+	/**
+	 * Add a timeout to the watchlist email block
+	 */
+	private function getTimeOutSql() {
+		global $wgEnableWatchlistNotificationTimeout, $wgWatchlistNotificationTimeout;
+
+		if ( !empty( $this->otherParam['notisnull'] ) ) {
+			$notificationTimeoutSql = "1";
+		} elseif ( !empty( $wgEnableWatchlistNotificationTimeout ) && isset( $WatchlistNotificationTimeout ) ) {
```

Comment:
Looks like we lost a ```wg``` on ```$WatchlistNotificationTimeout```.  Its funny, PHPStorm doesn't flag this for some reason, maybe because its being used within an ```isset```.  Maybe we can upgrade all of these to ```F::app()->wg->```